### ISSUE TITLE
[CLI] can't specify the artifact cache directory 

### ISSUE THREAD

**Issue by ornithos** [wandb/client#1912]
**Description**

The [documentation suggests](https://docs.wandb.ai/ref/run/init) that one can specify the artifact cache directory when initializing the run (i.e. `run.init(dir=...)`). 

>  dir (str, optional) 
>  An absolute path to a directory where metadata will be stored. When you call download() on an artifact, this is 
> the directory where downloaded files will be saved. By default this is the ./wandb directory.

However, it appears to me that the artifact directory is hardcoded to `./artifacts`, and the specified `dir` only changes the run metadata directory.  

A quick flick through the codebase for me suggested the [`get_artifacts_cache()`](https://github.com/wandb/client/blob/9238851ecdb33893490034a56478d3a064df1b03/wandb/sdk/interface/artifacts.py#L774) function which indeed does not appear to use the `run` directory. Is the documentation wrong, or is there some way around this (ideally which doesn't use environment variables)?

**Wandb features**

In case it's relevant, the artifact in question is stored on S3. I'm not using any other features of `wandb` other than `Artifact` for this example.

**How to reproduce**

Let's say I'm in directory `~/ml-model` (in all cases let's assume `~` is expanded prior to use, but is used here for clarity) and I run the following in Python:

```python
import wandb
run = wandb.init(dir="~/intended-wandb-directory")
artifact = run.use_artifact("username/project/artifact-name:latest", type="dataset")
artifact.download()
```

This results in

1. **Within `~/intended-wandb-directory`**: a `wandb` directory with the run metadata
2. **Within `~/ml-model`**: an `artifacts` directory with the subdirectory `artifact-name:latest`. I am hoping that this can be moved also within the `~/intended-wandb-directory`.

**Environment**
- OS: Ubuntu 20.04.2 LTS
- Environment: Jupyter Notebook (local)
- Python Version: 3.8.2
- `wandb` Version: 0.10.20

**Comment by vanpelt** [comment]
Hey @ornithos you can use the `WANDB_CACHE_DIR` environment variable to specify where the artifact data is cached.  By default it will be stored in `~/.cache/wandb`.

**Comment by ornithos** [comment]
Thanks @vanpelt. I did see this in the code, but this is at variance to the documentation (as above), which suggests this is possible natively using the python client kwargs. If environment variables are the only option at present, can I then propose that the documentation is updated accordingly?

**Comment by vanpelt** [comment]
@ornithos absolutely, that's why I'm not closing this issue.  We're doing a large overhaul of the logic we use to manage settings for all features of wandb.  We'll be sure this is properly documented in that project.

**Comment by cvphelps** [comment]
@ornithos I'd like to fix the docs in the meantime, could you please share where you found it? I'm not seeing this feature on the wandb.init page https://docs.wandb.ai/ref/run/init

**Comment by ornithos** [comment]
Thanks @cvphelps! Perhaps it's easiest to reference the line in the docstring: see [`wandb_init.py#L646`](https://github.com/wandb/client/blob/a8bc4f6f186c40caa3c708f6371d29edcf4c6b17/wandb/sdk/wandb_init.py#L646).

**Comment by diegoquintanav** [comment]
is this fixed? I'd be glad to know how to fix this :+1:

**Comment by vanpelt** [comment]
@diegoquintanav by default `artifact.download()` will download files into a directory named `./artifacts/artifatname:version`.  This way you can download multiple artifact versions without overwriting previous ones.  If you specify `root` the artifact will be download directly into that directory so you're responsible for cleaning it up.

This ticket is about specifying the cache directory which is a central caching directory that is used to prevent needing to re-download objects that already exist on your machine.

If you have an example of calling `artifact.download(root="my_dir")` and having files not endup in `my_dir` please create a new issue with the details about the wandb versions and code you used.

**Comment by diegoquintanav** [comment]
Hi @vanpelt, please check #2486. In any case, from the functionality described, I think that OP was referring to the artifacts folder and not the cache folder (at `~/.cache/wandb` for example). That folder has only md5 files as I see.  Moreover, if I specify root in download(), I won't have the same folder structure that I get _without_ specifying root. I think the functionality could use some work :v:
